STYLE_GUIDE.md

```diff
@@ -0,0 +1,25 @@
+# DELV Solidity Style Guide
+
+## Commenting
+
+1. Use complete sentences and end them with the appropriate puntuaction (typically a period).
+2. Use correct spelling.
+3. Use correct capitalization.
+4. Comments should not go past the 80th column from the start of the line.
+5. Add  full Natspec to each contract, library, function, storage variable, struct, event, and error.
+6. Within function bodies, code should be broken up into logical chunks (hereafter called "comment blocks") that are clearly commented. These comments should logically explain what the block is accomplishing. Any comments interleaved within blocks should be `NOTE` or `TODO` comments.
```

Comment:
> Any comments interleaved within blocks should be `NOTE` or `TODO` comments.

I'm a bit confused on what this is trying to convey... it reads to me like all comments within functions should be prepended with `NOTE` or `TODO`, but I feel like that isn't your intention.

Reply:
It’s trying to convey that the comment at the start of the block should be the only “naked” comment and that any comments within the block should be NOTE or TODO.

For example, this follows this:

```
// blah blah blah
uint256 a = foo();
a += b - c;
// NOTE: We have to reduce a because…
a = a.mulDown(0.5e18);
```

Whereas this doesn’t:

```
// blah blah blah
uint256 a = foo();
a += b - c;
// We have to reduce a because…
a = a.mulDown(0.5e18);
```

Reply:
It just helps you visually see the start of the code block by looking for the starting comment. Then it’s obvious that any comments inside the block are at a lower-level of detail.

Reply:
If this clears it up, I can just include this example. My thought was that we could add examples wherever they were needed, but I wasn’t going to do that for all of the rules because this is the minimum crappy style guide